include/exadg/fluid_structure_interaction/precice/coupling_surface.h

```diff
@@ -0,0 +1,235 @@
+#pragma once
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/vectorization.h>
+
+#include <deal.II/matrix_free/matrix_free.h>
+
+#include <exadg/matrix_free/integrators.h>
+#include <precice/SolverInterface.hpp>
+
+namespace ExaDG
+{
+namespace preCICE
+{
+using namespace dealii;
+
+/**
+ * Enum to handle all implemented data write methods one can use
+ */
+enum class WriteDataType
+{
+  undefined,
+  values_on_dofs,
+  values_on_other_mesh,
+  gradients_on_other_mesh,
+  values_on_q_points,
+  normal_gradients_on_q_points
+};
+
+/**
+ * A pure abstract base class, which defines the methods for the functions
+ * used in the main Adapter class. Each instance of all derived classes are
+ * always dedicated to a specific coupling mesh, i.e., the vertices used for
+ * the coupling. The instantiated objects provide functions on how to read
+ * and write data on this mesh and how to define the mesh by means of its
+ * spatial coordinates.
+ */
+template<int dim, int data_dim, typename VectorizedArrayType>
+class CouplingSurface
```

Comment:
I know, but the name 'Interface' was already rejected by @nfehn. What do you propose?

Reply:
I think I complained about "interface" because I thought you used it in the sense of interface-class (C++), and this could lead to misunderstandings. Maybe we have to think about what this class does or what its main responsibility is in order to find a name that all are happy with. I think "Base" inside the name would make sense to highlight that this class is used as a base class, something like `preCICE::InterfaceCouplingBase` or just `preCICE::CouplingBase` with derived classes `preCICE::DofCoupling`, `preCICE::QuadCoupling` (for example) would also be fine for me.

Reply:
This sounds reasonable, I will change it.